addon/components/groupedit-toolbar.js

```diff
@@ -1,148 +1,162 @@
-    this.get('groupEditEventsService').on('olvRowsDeleted', this, this._rowsDeleted);
-  },
+    @property createNewButton
+    @type Boolean
+    @default true
+  */
+  createNewButton: true,
 
   /**
-   * Implementation of component's teardown.
-   *
-   * @method willDestroy
-   */
-  willDestroy() {
-    this.get('groupEditEventsService').off('olvRowSelected', this, this._rowSelected);
-    this.get('groupEditEventsService').off('olvRowsDeleted', this, this._rowsDeleted);
-    this._super(...arguments);
-  },
+    Boolean property to show or hide delete button in toolbar.
+    Delete record button will not display if set to false.
+
+    @property deleteButton
+    @type Boolean
+    @default true
+  */
+  deleteButton: true,
 
   /**
-   * The collection of functions that will be invoked when
-   * click on toolbar buttons.
-   *
-   * @property actions
-   * @type Object
-   * @readOnly
-   */
+    The collection of functions that will be invoked when
+    click on toolbar buttons.
+
+    @property actions
+    @type Object
+    @readOnly
+  */
   actions: {
-    // Add record click button handler
-    addRow: function() {
+    /**
+      Handles add record button click and triggers add record event on
+      {{#crossLink "FlexberryGroupeditComponent"}}{{/crossLink}}.
+
+      @method actions.addRow
+    */
+    addRow() {
       if (this.get('readonly')) {
         return;
       }
 
       var componentName = this.get('componentName');
-      this.get('groupEditEventsService').addRowTrigger(componentName);
+      this.get('_groupEditEventsService').addRowTrigger(componentName);
     },
 
-    // Delete records click button handler
-    deleteRows: function() {
+    /**
+      Handles delete records button click and triggers delete records event on
+      {{#crossLink "FlexberryGroupeditComponent"}}{{/crossLink}}.
+
+      @method actions.deleteRows
+    */
+    deleteRows() {
       if (this.get('readonly')) {
         return;
       }
 
       var componentName = this.get('componentName');
-      this.get('groupEditEventsService').deleteRowsTrigger(componentName);
+      this.get('_groupEditEventsService').deleteRowsTrigger(componentName);
+    }
+  },
+
+  /**
+    The method called when component is instantiated.
+
+    @method init
+    @throws {Error} An error occurred during the initialization of component.
+  */
+  init() {
+    this._super(...arguments);
+    var componentName = this.get('componentName');
+    if (!componentName) {
+      throw new Error('Name of flexberry-groupedit component was not defined.');
     }
+
+    this.get('_groupEditEventsService').on('olvRowSelected', this, this._rowSelected);
+    this.get('_groupEditEventsService').on('olvRowsDeleted', this, this._rowsDeleted);
   },
 
   /**
-   * Boolean flag to indicate enabled state of delete rows button.
-   *
-   * @property isDeleteRowsEnabled
-   * @type Boolean
-   */
-  isDeleteRowsEnabled: undefined,
+    Implementation of component's teardown.
```

Comment:
Override to implement teardown.
For more information see [willDestroy](http://emberjs.com/api/classes/Ember.Component.html#method_willDestroy) method of [Ember.Component](http://emberjs.com/api/classes/Ember.Component.html).